src/consensus/encode.rs

```diff
@@ -192,17 +192,32 @@ pub fn serialize_hex<T: ?Sized>(data: &T) -> String
 pub fn deserialize<'a, T>(data: &'a [u8]) -> Result<T, Error>
      where T: Decodable<Cursor<&'a [u8]>>
 {
-    let mut decoder = Cursor::new(data);
-    let rv = Decodable::consensus_decode(&mut decoder)?;
+    let mut consumed: u64 = 0;
+    let rv = deserialize_partial(data, &mut consumed)?;
 
     // Fail if data is not consumed entirely.
-    if decoder.position() == data.len() as u64 {
+    if consumed == data.len() as u64 {
         Ok(rv)
     } else {
+        eprintln!("total {}, consumed {}", data.len(), consumed);
```

Comment:
Oh, I see the other motivation is to know the number of bytes consumed in from stream.

Reply:
It would be better to pass Cursor to deserialize_partial instead of rewriting its logic with a
```
let mut consumed = 0 
```
also passing &mut consumed id rather C/C++ style where you are restricted to a single return value.

the type usize is meant to be handling indices, it has the appropriate size for the OS memory model. Do not use u64 for that.

Reply:
Completely reworked in #231 as you suggested: no &mut parameter, using Cursor etc. Thanks for pointing all that out.